### PR TITLE
PT-3186: Improve advanced data filtering form

### PR DESCRIPTION
**Pull request by marta-**

### CODE REVIEW

**Comment by sashaandjic** [comment]
If we shorten text, we may be able to display options on their own separate line. For example:
Filter records with:
all
at least one diagnosis

all
at least one phenotype

all subcategories
exact phenotypes

Candidate genes
Confirmed Causative genes

The other suggestion is to italicize the text so it is consistent with format of "after" and "before"?

**Comment by sashaandjic** [comment]
It appears a bit cluttered when having long and multiple disease/phenotype descriptions. Would it be possible to display only the part of the description that fits on one line with a tooltip showing the full text?
new filters
old

**Comment by marta-** [comment]
@sashaandjic Changed long names as per your suggestion in latest commit. Not sure I agree with displaying filter options one per line, I personally don't mind the current look in the side filters (please note i fixed the positioning of "Delete all") and i think one per line would look bad in the export/push ui. Other opinions are welcome.

**Comment by sashaandjic** [comment]
"Delete all" option is somewhat difficult to click on

**Comment by sashaandjic** [comment]
under Phenotypes replace "Filter records O with all subcategories of O only matching exactly the selected phenotypes." with "Filter records that match the selected phenotypes O exactly O including subcategories"

**Comment by marta-** [comment]
@sashaandjic Forgot to mention: I only tested my changes in Chrome & FF. For all I know it all might look like a train wreck in IE & Safari. Please make sure you or someone else tests in those browsers too.

**Comment by marta-** [comment]
@sashaandjic Comments addressed.

**Comment by sashaandjic** [comment]
looks very good! few minor complains:

**Comment by marta-** [comment]
@sashaandjic :
1 and 2 are not specific to filters, that happens everywhere in PhenoTips. Please log them separately.
3: maybe we should have kept the "empty means no constraints" hints?

**Comment by marta-** [comment]
@sashaandjic @veronikaslc @katpavlova did anyone test in Safari & IE?

**Comment by sashaandjic** [comment]
sure...oh there is one more that I forgot: if I select more than one owner(referer) it will not return any records as it applies "AND" between them. Meanwhile, everywhere else OR is applied.

**Comment by sashaandjic** [comment]
In Safari check boxes and radio buttons appear unaligned with their respective labels
screen shot 2017-07-06 at 12 20 10 pm

**Comment by marta-** [comment]
@sashaandjic 
> if I select more than one owner(referer) it will not return any records as it applies "AND" between them. Meanwhile, everywhere else OR is applied.

The referrer seems to work fine, can you confirm there are problems only with the owner filter?

The issue with the owner is not that the values are AND-ed, but that when there's more than one owner value we're searching for local references (e.g. `XWiki.Admin`, `XWiki.jdoe`) while storing full references (`xwiki:XWiki:Admn`). This bug needs to be logged separately, as it affects the stable branch 1.3.x as well.

**Comment by sashaandjic** [comment]
confirming the problem with the owner; by referrer and by last author are fine
screen shot 2017-07-06 at 2 30 35 pm

**Comment by veronikaslc** [comment]
1. I still think we can use placeholders for dates "before:" and "after:", it's not that hard to learn
2. There is no way I can install windows 9 on my machine to test, I have downloaded IETester that allows to test in multiple versions of IE and all of them worked fine except IE9 that crashed on the page.

**Comment by sashaandjic** [comment]
In Safari I noticed that radio buttons on group and study filters look much worse than on the patient and family filters (probably because they are thinner).
group filter
patient filter

**Comment by sashaandjic** [comment]
it not the width, group and study filters have bullet points that are causing this anomaly.

**Comment by sashaandjic** [comment]
One more thing, the date fields on groups and study filters need to be done as in the patient filter. 

Group/study filter:
dates on groups and studies

**Comment by marta-** [comment]
@sashaandjic the last two issues are actually related to (and fixed by) #2346. I should rebase, since that one is already merged into master.

**Comment by sashaandjic** [comment]
ok

**Comment by veronikaslc** [comment]
should we re-review?

**Comment by marta-** [comment]
@veronikaslc yes, please

File: components/navigation/ui/src/main/resources/PhenoTips/DBWebHomeSheet.xml [hunk 1]
```diff
@@ -374,21 +377,144 @@ document.observe('xwiki:livetable:newrow', function(tableEvent){
...
+.entity-directory .xwiki-livetable-topfilters-tip .tippointer {
+  display: none;
+}
+
+/*---------------------------------------------------------------*/
+.entity-directory &gt; .xwiki-livetable-container.hidden-filters {
+    margin-left: 20px;
+}
+.entity-directory .hidden-filters .tipfilters {
+    xtransform: rotate(-90deg);
```
Comment (sdumitriu) [inline components/navigation/ui/src/main/resources/PhenoTips/DBWebHomeSheet.xml]: Trash?

Comment (marta-) [inline components/navigation/ui/src/main/resources/PhenoTips/DBWebHomeSheet.xml]: yep, sorry, removing.

File: components/navigation/ui/src/main/resources/PhenoTips/DatabaseFilters.xml [hunk 2]
```diff
@@ -51,41 +51,41 @@
...
+* $services.localization.render('phenotips.databaseFilters.externalID') {{html clean=false}}&lt;input type="text" name="external_id" value=""/&gt;{{/html}}
 #end
 #if (!$xwiki.getDocument('PhenoTips.VisibilityClass').isNew())
-  * $services.localization.render('phenotips.databaseFilters.visibility') {{html clean=false}}#foreach ($visibility in $services.permissions.listVisibilityOptions())&lt;br /&gt;&lt;input type="checkbox" class="multipleSelection privacyExport" name="visibility" value="$visibility.name"#if ($visibility.name != 'hidden') checked="checked"#end /&gt; $visibility.label#end&lt;input type="hidden" name="visibility/class" value="PhenoTips.VisibilityClass"/&gt;{{/html}}
+  * $services.localization.render('phenotips.databaseFilters.visibility') {{html clean=false}}#foreach ($visibility in $services.permissions.listVisibilityOptions())&lt;br /&gt;&lt;label&gt;&lt;input type="checkbox" class="multipleSelection privacyExport" name="visibility" value="$visibility.name"#if ($visibility.name != 'hidden') checked="checked"#end /&gt; $visibility.label&lt;label&gt;#end&lt;input type="hidden" name="visibility/class" value="PhenoTips.VisibilityClass"/&gt;{{/html}}
```
Comment (sdumitriu) [inline components/navigation/ui/src/main/resources/PhenoTips/DatabaseFilters.xml]: /label (you forgot the `/`)

**Pull request merged**